### ISSUE TITLE
Rename data filter properties

### ISSUE THREAD

**Issue by cdavernas** [serverlessworkflow/specification#672]
**What would you like to be added**:

To rename data filter properties

**Why is this needed**:

Having properties such as `stateDataFilter` when defining a state, or `actionDataFilter` when defining an action is redudant and tedious to write. So, instead of:

```yaml
...
states:
- name: MyOperationState
  type: operation
  stateDataFilter: ...
  actions:
    - name: MyAction
      actionDataFilter: ...
      ...
...
```

... I propose we use:

```yaml
...
states:
- name: MyOperationState
  type: operation
  dataFilter: ...
  actions:
    - name: MyAction
      dataFilter: ...
      ...
...
```

IMHO it's less confusing, shorter and cleaner.

**Comment by tsurdilo** [comment]
agree but much harder to define cleanly in jaon schema :) looking into it to see if there is nice way

**Comment by cdavernas** [comment]
Closed as resolved by 0.10, and therefore as part of #843
